### ISSUE TITLE
[TW-1931] wait:<year> does not work as expected

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#1943]
_Michele Ippolito on 2017-10-18T06:40:58Z says:_ 

 I issued the following command:

 

    task 1 2 3 modify wait:2018

 

I expected the wait date to be set to January 1st of next year.

Instead, I get this output:
*  Wait will be set to '1970-01-01'.

 

 

I can also reproduce this with any other year.

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2017-10-18T06:40:58Z
Modified: 2017-10-18T09:12:10Z
```

**Comment by taskwarrior** [comment]
_Dirk Deimeke on 2017-10-18T08:52:49Z says:_ 

 As indicated in the manpage, wait takes a date (not a year) as parameter.

So, you can easily use one of the following terms:
{code:java}
task 1 2 3 modify wait:2018-01-01 # ISO works everytime, you could use your own dateformat
task 1 2 3 modify wait:soy        # soy is a shortcut for "start of (next) year"
task 1 2 3 modify wait:soq        # soq is a shortcut for "start of (next) quarter"
{code}

**Comment by taskwarrior** [comment]
_Dirk Deimeke on 2017-10-18T08:54:00Z says:_ 

 Please be aware that the recent version of Taskwarrior is 2.5.1

**Comment by taskwarrior** [comment]
_Dirk Deimeke on 2017-10-18T08:54:26Z says:_ 

 This is not a bug.

**Comment by taskwarrior** [comment]
_Michele Ippolito on 2017-10-18T08:59:42Z says:_ 

 Sure, I know about the date shortcuts or ISO. The thing is that wait doesn't necessarily take a date, it can take a time too.

When I wrote a year, I expected one of two things:
 * it would assume it was a year (2018 is not a valid time in HH:MM format, nor it's a shortcut like "soy") and use 2018-01-01;
 * it would not recognize it and give me an error of some sort.

Instead, it appears to silently default to the linux epoch. This is surprising (and if I had not bothered to read the output, I'd have accepted a wrong date). IMHO, it should detect an invalid date and refuse to continue.

Unfortunately 2.5.1 is not on cygwin yet (I'll try to install it from source when I can find some time to try it).

**Comment by taskwarrior** [comment]
_Michele Ippolito on 2017-10-18T09:01:26Z says:_ 

 It already recognizes invalid wait arguments, for example:
{code:java}
~$ task 1 modify wait:dirk
'dirk' is not a valid date in the 'Y-M-D' format.
{code}
I expected the same thing to happen with an invalid date.

**Comment by taskwarrior** [comment]
_Dirk Deimeke on 2017-10-18T09:06:28Z says:_ 

 I get your point

Taskwarrior 2.6.0:

 
{code:java}
$ task add due:2018 test
'2018' is not a valid date in the '' format.
task 4 mod wait:2018
Modifying task 4 'xxx'.
'2018' is not a valid date in the '' format.
{code}
I am running a development snapshot of Taskwarrior.

**Comment by taskwarrior** [comment]
_Dirk Deimeke on 2017-10-18T09:07:25Z says:_ 

 Please note that Cygwin is not a supported platform anymore and maybe use the Windows Subsystem for Linux on Windows 10 instead.

**Comment by taskwarrior** [comment]
_Michele Ippolito on 2017-10-18T09:12:10Z says:_ 

 I got access to a Ubuntu box with taskwarrior 2.5.0 and I confirm it is already fixed in 2.5.0.

> maybe use the Windows Subsystem for Linux on Windows 10 instead.

Until my sysadmin decides to upgrade me to CE, gotta make do with what I have ;)
